metrics/cli/lib/doctor/doctor.dart

```diff
@@ -0,0 +1,57 @@
+// Use of this source code is governed by the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+import 'package:cli/cli/firebase/firebase_command.dart';
+import 'package:cli/cli/git/git_command.dart';
+import 'package:cli/common/model/services.dart';
+import 'package:cli/flutter/service/flutter_service.dart';
+import 'package:cli/gcloud/service/gcloud_service.dart';
+
+/// A class that provides an ability to check whether all required third-party
+/// services are available and get their version.
+class Doctor {
+  /// A service that provides methods for working with Flutter.
+  final FlutterService _flutterService;
+
+  /// A service that provides methods for working with GCloud.
+  final GCloudService _gcloudService;
+
+  /// A class that provides methods for working with the Firebase.
+  final FirebaseCommand _firebaseCommand;
+
+  /// A class that provides methods for working with the Git.
+  final GitCommand _gitCommand;
+
+  /// Creates a new instance of the [Doctor] with the given services.
+  ///
+  /// Throws an [ArgumentError] if the given [services] is `null`.
+  /// Throws an [ArgumentError] if the given [firebaseCommand] is `null`.
+  /// Throws an [ArgumentError] if the given [gitCommand] is `null`.
+  Doctor({
+    Services services,
+    FirebaseCommand firebaseCommand,
+    GitCommand gitCommand,
+  })  : _gcloudService = services?.gcloudService,
+        _flutterService = services?.flutterService,
+        _firebaseCommand = firebaseCommand,
+        _gitCommand = gitCommand {
+    ArgumentError.checkNotNull(services, 'services');
+    ArgumentError.checkNotNull(_firebaseCommand, 'firebaseCommand');
+    ArgumentError.checkNotNull(_gitCommand, 'gitCommand');
+  }
+
+  /// Checks versions of the required third-party services.
+  Future<void> checkVersions() async {
+    await _tryCatch(_flutterService.version);
+    await _tryCatch(_firebaseCommand.version);
+    await _tryCatch(_gcloudService.version);
+    await _tryCatch(_gitCommand.version);
+  }
+
+  /// Executes a [function] and catches errors if any.
+  Future<void> _tryCatch(Function function) async {
```

Comment:
```suggestion
  /// Checks version of the third-party service using the given
  /// [versionCallback].
  ///
  /// Catches all thrown exceptions to be able to proceed with checking the
  /// version of all the rest services.
  Future<void> _checkVersion(Future<void> Function() versionCallback) async {
```
What do you think? Also, I believe it would be great to change this method to make it accept the `InfoService` as a parameter once we'll implement all required services.